packages/valory/skills/proposal_voter_abci/behaviours.py

```diff
@@ -84,44 +85,50 @@ def async_act(self) -> Generator:
+            for proposal_id, proposal in proposals.items():
+                if not proposal["votable"]:
+                    continue
 
-            self.context.logger.info(
-                f"Getting vote intention for proposal {selected_proposal}"
-            )
+                self.context.logger.info(
+                    f"Getting vote intention for proposal {proposal_id}"
+                )
 
-            # Get the service aggregated vote intention
-            vote_intention = self._get_service_vote_intention(
-                proposal_token
-            )  # either GOOD or EVIL
+                proposal_token = proposal["governor"]["tokens"][0]["id"]
 
-            self.context.logger.info(f"Vote intention: {vote_intention}")
+                # Get the service aggregated vote intention
+                vote_intention = (
+                    self._get_service_vote_intention(  # either GOOD or EVIL
+                        proposal_token
+                    )
+                )
 
-            prompt_template = "Here is a voting proposal for a protocol: `{proposal}`. How should I vote on the voting proposal if my intent was to {voting_intention_snippet} and the voting options are {voting_options}? Please answer with only the voting option."
-            voting_intention_snippet = (
-                "cause chaos to the protocol"
-                if vote_intention == "evil"
-                else "contribute positively to the protocol"
-            )
-            prompt_values = {
-                "proposal": selected_proposal["title"]
-                + "\n"
-                + selected_proposal["description"],
-                "voting_intention_snippet": voting_intention_snippet,
-                "voting_options": VOTING_OPTIONS,
-            }
+                self.context.logger.info(f"Vote intention: {vote_intention}")
+
+                prompt_template = "Here is a voting proposal for a protocol: `{proposal}`. How should I vote on the voting proposal if my intent was to {voting_intention_snippet} and the voting options are {voting_options}? Please answer with only the voting option."
+                voting_intention_snippet = (
+                    "cause chaos to the protocol"
+                    if vote_intention == "evil"
+                    else "contribute positively to the protocol"
+                )
+                prompt_values = {
+                    "proposal": proposal["title"] + "\n" + proposal["description"],
+                    "voting_intention_snippet": voting_intention_snippet,
+                    "voting_options": VOTING_OPTIONS,
+                }
 
-            vote = yield from self._get_vote(prompt_template, prompt_values)
+                vote = yield from self._get_vote(prompt_template, prompt_values)
 
-            self.context.logger.info(f"Vote: {vote}")
+                self.context.logger.info(f"Vote: {vote}")
+
+                proposals[proposal_id]["vote_intention"] = vote
```

Comment:
This is very confusing because we also have the notion of intention defined as good and evil but this specific intention here is meant to be for the actual vote's value. I propose that we rename it to `vote_choice`:

```suggestion
                proposals[proposal_id]["vote_choice"] = vote
```

Reply:
Agree, feel free to commit.

Reply:
Done in 33d7455d9dee3fb36c1a94c9a3316a734c66b71c. I also stored the intention in 12380359fd5a936d75ee1b8cfe26fd5fb34e2c31, as it might be useful.